lumigator/python/mzai/schemas/experiments.py

```diff
@@ -16,21 +16,17 @@ class ExperimentCreate(BaseModel):
     system_prompt: str | None = None
     config_template: str | None = None
 
+class ExperimentResult(BaseModel):
```

Comment:
this would also change right? https://github.com/mozilla-ai/lumigator/blob/6b89a1520ead58b15b04ca2f5eefe44d851e9146/lumigator/python/mzai/backend/services/experiments.py#L178

Reply:
Yes, of course. The server impl will need to be adjusted.

Note that, in this case, we're leaking the FK solution in the DB into the API, which is not desirable because using two tables with FKs is actually an implementation aspect of our 1:1 relation between experiments and results (and since it's a 1:1, one could actually think about storing the results inside the experiment itself, but as it is we have the possibility to change to 1:M later on, if an experiment provides several results).

Reply:
_Actually_ one could think why a join is not done in this case. The original is (crossing class boundaries, etc etc):

```
record = self.experiment_repo.get(experiment_id)
...
result_record = self.result_repo.get_by_experiment_id(experiment_id)
```

Which I'd assume to result in two separate SQL statements. Maybe we could provide another method that combines both like this:

```
    def get_by_experiment_id(self, experiment_id: UUID) -> ExperimentResultRecord | None:
        return (
            self.session.query(ExperimentRecord, ExperimentResultRecord)
            .where(ExperimentResultRecord.experiment_id == experiment_id)
            .where(ExperimentRecord.experiment_id == ExperimentResultRecord.experiment_id)
            .first()
        )
```

This ensures transactional behaviour and can also be reused later on to retrieve sets of experiments+results (by dropping the first `where`).